dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/collection/CollectionUtils.java

```diff
@@ -425,4 +426,29 @@ public static <K, V> Map<K, V> merge(@Nonnull Map<K, V> m1, @Nonnull Map<K, V> m
     merged.putAll(m2);
     return Map.copyOf(merged);
   }
+
+  /**
+   * Combines 2 Lists into a memory-efficient unmodifiable view of both. Safe to perform read/get
+   * operations on. Updates on original Lists are reflected in the returned view. Can't be modified.
+   *
+   * @param l1 list 1
+   * @param l2 list 2
+   * @return combined view of both lists
+   * @param <T> type
+   */
+  public static <T> List<T> combinedUnmodifiableView(List<T> l1, List<T> l2) {
```

Comment:
This seems not worth the trouble. There is so much more somebody might do with the result that is not correct yet. For exmaple, `contains`, `iterator`, `isEmpty`, ...
I know it is frustrating that the JDK does not have a concat view already. I tend to use `Stream.concat(a.stream(), b.stream()).toList()`. Sure that copies to a new list but for short lists this isn't a big deal. 
Alternatively be very clear about the limitations of this view in the javadoc.